wgengine/monitor/monitor_darwin.go

```diff
@@ -87,9 +89,44 @@ func (m *darwinRouteMon) Receive() (message, error) {
 }
 
 func (m *darwinRouteMon) skipMessage(msg route.Message) bool {
-	switch msg.(type) {
+	switch msg := msg.(type) {
 	case *route.InterfaceMulticastAddrMessage:
 		return true
+	case *route.InterfaceAddrMessage:
+		return m.skipInterfaceAddrMessage(msg)
+	case *route.RouteMessage:
+		return m.skipRouteMessage(msg)
+	}
+	return false
+}
+
+func (m *darwinRouteMon) skipInterfaceAddrMessage(msg *route.InterfaceAddrMessage) bool {
+	for i, a := range msg.Addrs {
```

Comment:
Can you look up the unix.RTAX_IFP offset directly, instead of looping until you find the right index? Need to check overall length, but seems more efficient otherwise.